pyodide-build/pyodide_build/pyzip.py

```diff
@@ -0,0 +1,128 @@
+import shutil
+from collections.abc import Callable
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+# This files are removed from the stdlib by default
+REMOVED_FILES = (
+    # package management
+    "ensurepip/",
+    "venv/",
+    # build system
+    "lib2to3/",
+    # other platforms
+    "_osx_support.py",
+    # Not supported by browser
+    "curses/",
+    "dbm/",
+    "idlelib/",
+    "tkinter/",
+    "turtle.py",
+    "turtledemo",
+    "webbrowser.py",
+)
+
+# This files are unvendored from the stdlib by default
+UNVENDORED_FILES = (
+    "test/",
+    "distutils/",
+    "sqlite3",
+    "ssl.py",
+    "lzma.py",
+)
+
+# TODO: These modules have test directory which we unvendors it separately.
+#       So we should not pack them into the zip file in order to make e.g. import ctypes.test work.
+#       Note that all these tests are moved to the subdirectory of `test` module in upstream CPython 3.12.0a1.
+#       So we don't need this after we upgrade to 3.12.0
+NOT_ZIPPED_FILES = ("ctypes/", "unittest/")
+
+
+def default_filterfunc(
+    root: Path, verbose: bool = False
+) -> Callable[[str, list[str]], set[str]]:
+    def filterfunc(path: Path | str, names: list[str]) -> set[str]:
+        filtered_files = {
+            (root / f).resolve()
+            for f in REMOVED_FILES + UNVENDORED_FILES + NOT_ZIPPED_FILES
+        }
+
+        path = Path(path).resolve()
+
+        if path.name == "__pycache__":
+            return set(names)
+
+        _names = []
+        for name in names:
+
+            fullpath = path / name
+
+            if fullpath.name == "__pycache__" or fullpath in filtered_files:
+                if verbose:
+                    print(f"Skipping {fullpath}")
+
+                _names.append(name)
+
+        return set(_names)
+
+    return filterfunc
+
+
+def create_zip(
+    libdir: Path | str,
+    output: Path | str = "python",
+    pycompile: bool = False,
+    filterfunc: Callable[[str, list[str]], set[str]] | None = None,
+) -> None:
+    """
+    Bundle Python standard libraries into a zip file.
+
+    The basic idea of this function is similar to the standard library's
+    [PyZipFile](https://github.com/python/cpython/blob/59665d0280c2299ea87e9af45cedc90656cb6f55/Lib/zipfile/__init__.py#L2011).
```

Comment:
```
{ref}`zipfile.PyZipFile`
``` should work I think once we enable stdlib sphinx inter-linking in https://github.com/pyodide/pyodide/pull/3405/files#diff-85933aa74a2d66c3e4dcdf7a9ad8397f5a7971080d34ef1108296a7c6b69e7e3R74